test/encryption_tests_v4.cpp

```diff
@@ -116,6 +117,38 @@ std::string GenerateUid2TokenV4AndValidate(
     return advertisingToken;
 }
 
+TEST(EncryptionTestsV4, CanDecryptV4TokenEncodedAsBase64)
+{
+    UID2Client client("ep", "ak", CLIENT_SECRET, IdentityScope::UID2);
+    client.RefreshJson(KeySetToJson({MASTER_KEY, SITE_KEY}));
+    auto advertisingToken = GenerateUid2TokenV4AndValidate(EXAMPLE_UID, MASTER_KEY, SITE_ID, SITE_KEY, EncryptTokenParams());
+
+    while (true) {
+        const bool hasBase64UrlChars = std::any_of(advertisingToken.begin(), advertisingToken.end(), [](char c) { return c == '-' || c == '_'; });
+        if (hasBase64UrlChars) {
+            break;
+        }
+        advertisingToken = GenerateUid2TokenV4AndValidate(EXAMPLE_UID, MASTER_KEY, SITE_ID, SITE_KEY, EncryptTokenParams());
+    }
+
+    const bool isBase64UrlEncoding = std::any_of(advertisingToken.begin(), advertisingToken.end(), [](char c) { return c == '-' || c == '_'; });
+    EXPECT_TRUE(isBase64UrlEncoding);
+
+    std::vector<std::uint8_t> adTokenBytes;
+    uid2::UID2Base64UrlCoder::Decode(advertisingToken, adTokenBytes);
```

Comment:
Is any of the below code duplicated from the other tests?
Worth making a function?

Reply:
done some refactoring to have DecryptAndAssertSuccess method like other SDK